python/lsst/daf/butler/_butler_repo_index.py

```diff
@@ -41,10 +43,14 @@
 class ButlerRepoIndex:
     """Index of all known butler repositories.
 
-    The index of butler repositories is found by looking for a
-    configuration file at the URI pointed at by the environment
-    variable ``$DAF_BUTLER_REPOSITORY_INDEX``. The configuration file
-    is a simple dictionary lookup of the form:
+    The index of butler repositories can be configured in two ways:
+    1. By setting the environment variable ``DAF_BUTLER_REPOSITORY_INDEX`` to
+    the URI of a configuration file.
+    2. By setting the environment variable ``DAF_BUTLER_REPOSITORIES`` to the
+    contents of the configuration file as a string.
```

Comment:
If you are wanting this to be YAML you should document that here. Are you sure you don't want to require it to be JSON so that you don't have to have embedded newlines and it can parse a little faster? `DAF_BUTLER_REPOSITORY_INDEX` can work with JSON files as well.

Reply:
In the real deployment I am passing it as JSON, but I thought it was less confusing to have the same file format for both instead of trying to document "well if you do it directly it has to be JSON, but if it's the other one it can be YAML or JSON".